test/tstQueryTreeCallbackKernelQuery.cpp

```diff
@@ -0,0 +1,71 @@
+/****************************************************************************
+ * Copyright (c) 2023 by the ArborX authors                                 *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the ArborX library. ArborX is                       *
+ * distributed under a BSD 3-clause license. For the licensing terms see    *
+ * the LICENSE file in the top-level directory.                             *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+#include "ArborX_EnableDeviceTypes.hpp" // ARBORX_DEVICE_TYPES
+#include <ArborX.hpp>
+
+#include <boost/test/unit_test.hpp>
+
+#include <numeric>
+#include <vector>
+
+BOOST_AUTO_TEST_SUITE(KerneliQueryCallbacks)
+
+struct IntersectionCallback
+{
+  int query_index;
+  bool &success;
+
+  template <typename Query, typename Value>
+  KOKKOS_FUNCTION void operator()(Query const &, Value const &value) const
+  {
+    ArborX::Box actual_box = value.bounding_volume;
+    ArborX::Point min = actual_box.minCorner();
+    if (query_index != min[0] || query_index != min[1] || query_index != min[2])
+      success = false;
+    else
+      success = true;
+  }
```

Comment:
This seems fragile. Why not do
```suggestion
    success = (query_index == value.index);
```

Reply:
That's fair. Changed.